pkg/manager/internal/controllers/organizationrole_controller.go

```diff
@@ -0,0 +1,178 @@
+		organizationRole.Status.ObservedGeneration = organizationRole.Generation
+		organizationRole.Status.SetCondition(corev1alpha1.OrganizationRoleCondition{
+			Type:    corev1alpha1.OrganizationRoleReady,
+			Status:  corev1alpha1.ConditionTrue,
+			Reason:  "SetupComplete",
+			Message: "OrganizationRole setup is complete.",
+		})
+		isChanged = true
+	}
+	if isChanged {
+		if err := r.Status().Update(ctx, organizationRole); err != nil {
+			return ctrl.Result{}, fmt.Errorf("updating OrganizationRole status: %w", err)
+		}
+	}
+
+	return ctrl.Result{}, nil
+}
+
+func (r *OrganizationRoleReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	enqueueAllRoles := &handler.EnqueueRequestsFromMapFunc{
+		ToRequests: handler.ToRequestsFunc(func(mapObject handler.MapObject) (out []ctrl.Request) {
+			organizationRoles := &corev1alpha1.OrganizationRoleList{}
+			if err := r.Client.List(context.Background(), organizationRoles); err != nil {
+				// This will makes the manager crashes, and it will restart and reconcile all objects again.
+				panic(fmt.Errorf("listting OrganizationRoles: %w", err))
+			}
+			for _, organizationRole := range organizationRoles.Items {
+				out = append(out, ctrl.Request{
+					NamespacedName: types.NamespacedName{
+						Name:      organizationRole.Name,
+						Namespace: organizationRole.Namespace,
+					},
+				})
+			}
+			return
+		}),
+	}
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&corev1alpha1.OrganizationRole{}).
+		Owns(&rbacv1.Role{}).
+		Watches(&source.Kind{Type: &corev1alpha1.OrganizationRoleTemplate{}}, enqueueAllRoles).
+		Complete(r)
+}
+
+func (r *OrganizationRoleReconciler) handleDeletion(ctx context.Context, log logr.Logger, organizationRole *corev1alpha1.OrganizationRole) error {
+	// Update the OrganizationRole Status to Terminating.
+	readyCondition, _ := organizationRole.Status.GetCondition(corev1alpha1.OrganizationRoleReady)
+	if readyCondition.Status != corev1alpha1.ConditionFalse ||
+		readyCondition.Status == corev1alpha1.ConditionFalse && readyCondition.Reason != corev1alpha1.OrganizationRoleTerminatingReason {
+		organizationRole.Status.ObservedGeneration = organizationRole.Generation
+		organizationRole.Status.SetCondition(corev1alpha1.OrganizationRoleCondition{
+			Type:    corev1alpha1.OrganizationRoleReady,
+			Status:  corev1alpha1.ConditionFalse,
+			Reason:  corev1alpha1.OrganizationRoleTerminatingReason,
+			Message: "OrganizationRole is being terminated",
+		})
+		if err := r.Status().Update(ctx, organizationRole); err != nil {
+			return fmt.Errorf("updating OrganizationRole status: %w", err)
+		}
+	}
+	return nil
+}
+
+func (r *OrganizationRoleReconciler) reconcileRole(ctx context.Context, organizationRole *corev1alpha1.OrganizationRole) (acceptedRules []rbacv1.PolicyRule, err error) {
+	organizationRoleTemplates := &corev1alpha1.OrganizationRoleTemplateList{}
+	if err = r.List(ctx, organizationRoleTemplates); err != nil {
+		return
+	}
+	var maxRules []rbacv1.PolicyRule
```

Comment:
I'd disagree with the rule cap being implemented like this. Instead, it should probably be defined in the `Organization` CRD itself?

Reply:
Sorry I did't get it :) 
What do you mean by defining in `Organization`?

Reply:
By defining it in the `Organization` I mean we have a field `.spec.maximumPermissions` or something like it with rule lists which are the maximum allowed policy rules allowed within this organization.

Reply:
We cannot pre-define the maximum permission of the Organization in the Organization itself. On one knows which permission will be needed when they create the `Organization`, for example, when we integrate Bulward with KubeCarrier, everything is dynamic, if `couchdb` objects are created, `Organization` should grant with `couchdb` related permissions.

Reply:
Where do we define it then? @thetechnick 

It needs to be somewhere nicely defined and specified for ease of use.

Reply:
So we define it in  the `OrganizationRoleTemplate`, and then give the operator/admin for example, could be KubeCarrier operator to update the `OrganizationRoleTemplate` to grant/revoke permissions, and every time organization admin create the new `OrganizationRole` we check it with the Max Rule, like what is implemented now, and accepted rule will be shown in the status of `OrganizationRole`

Reply:
This isn't the obvious way. Where can I find out what's the maximum role scope for this organization? It's hidden away in OrganizationRoleTemplates. 

Also, nothing about this is mentioned in the proposal IIRC.

Reply:
It was mentioned a bit here:https://github.com/kubermatic/bulward/pull/66/files#diff-da75ad503782278090a6e6ef7b58c9d7R9-R10

I am not sure if user should care about the maximum permission at all, at least Organization Owner will have maximum permission.

Reply:
ohh,, now I see it; it's a bit like a footnote.

>  This maximum permission level results from all roles that are assigned to the organization unit and thus to the Organization/Project Owner.

How bout we talk it briefly in tomorrow's daily? 

---

we should at least aggregate maximum permission cap somewhere, like in the Organization status, to have a clear picture what it is.